### PR TITLE
Wait on futures in preemptibles test

### PR DESCRIPTION
**Pull request by rtitle**
I think this was probably causing https://github.com/DataBiosphere/leonardo/issues/204. Our calls to `googleStorageDAO.copyObject` and `googleStorageDAO.setObjectAccessControl` were racy, which probably made the latter fail intermittently, which manifested as 403 errors in a notebook.

Note we've encountered some other errors related to this test (see https://github.com/DataBiosphere/leonardo/issues/228). I haven't been able to reproduce those locally, I'm going to run it a bunch of times in Jenkins.

Have you read [CONTRIBUTING.md](../CONTRIBUTING.md) lately? If not, do that first.

I, the developer opening this PR, do solemnly pinky swear that:

- [ ] I've documented my API changes in Swagger

In all cases:

- [ ] Get two thumbsworth of review and PO signoff if necessary
- [ ] Verify all tests go green
- [ ] Squash and merge; you can delete your branch after this
- [ ] Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

**Comment by rtitle** [comment]
1 pass
jenkins retest

**Comment by rtitle** [comment]
2 passes
Apparently jenkins retest can be a substring

**Comment by rtitle** [comment]
3 passes
jenkins retest

**Comment by rtitle** [comment]
4th run showed a different intermittent failure:
https://fc-jenkins.dsp-techops.broadinstitute.org/job/leonardo-fiab-test-runner/465/testReport/junit/org.broadinstitute.dsde.workbench.leonardo/ClusterMonitoringSpec/Leonardo_clusters_should_execute_Hail_with_correct_permissions_on_a_cluster_with_preemptible_workers/

Looks like a timeout waiting on this line: https://github.com/DataBiosphere/leonardo/blob/rt-fix-preemptible-test-futures/automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/LeonardoTestUtils.scala#L346

The timeout is currently set to 1 minute. These Hail calls can take a while, maybe I will increase the timeout in this case.

**Comment by rtitle** [comment]
5: pass
jenkins retest

**Comment by rtitle** [comment]
6: pass
jenkins retest

**Comment by rtitle** [comment]
7: pass
jenkins retest

**Comment by rtitle** [comment]
8: pass 
jenkins retest
will stop at 10

**Comment by rtitle** [comment]
9: pass 
jenkins retest

**Comment by rtitle** [comment]
10: fail!
```
 The code passed to eventually never returned normally. Attempted 100 times over 30.163475045533335 minutes. Last failure message: List(Running, Error) did not contain element Creating. (LeonardoTestUtils.scala:109)
```
A cluster didn't finish creating after 30 minutes. :( Will investigate this.

**Comment by rtitle** [comment]
Couldn't get to the Google bucket for the above failure, but https://github.com/DataBiosphere/leonardo/pull/245 should help with that.
jenkins retest

**Comment by rtitle** [comment]
jenkins retest

**Comment by rtitle** [comment]
jenkins retest, I think some jenkins nodes are bad

**Comment by rtitle** [comment]
jenkins retest

File: automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/ClusterMonitoringSpec.scala [hunk 1]
```diff
@@ -110,14 +110,16 @@ class ClusterMonitoringSpec extends FreeSpec with LeonardoTestUtils with Paralle
         Orchestration.billing.addUserToBillingProject(projectName, ronEmail, Orchestration.billing.BillingProjectRole.User)(hermioneAuthToken)
 
         withNewGoogleBucket(project) { bucket =>
+          implicit val patienceConfig: PatienceConfig = storagePatience
+
           val srcPath = parseGcsPath("gs://genomics-public-data/1000-genomes/vcf/ALL.chr20.integrated_phase1_v3.20101123.snps_indels_svs.genotypes.vcf").right.get
           val destPath = GcsPath(bucket, GcsObjectName("chr20.vcf"))
-          googleStorageDAO.copyObject(srcPath.bucketName, srcPath.objectName, destPath.bucketName, destPath.objectName)
+          googleStorageDAO.copyObject(srcPath.bucketName, srcPath.objectName, destPath.bucketName, destPath.objectName).futureValue
```
Comment (jmthibault79) [inline automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/ClusterMonitoringSpec.scala]: oh no
picard face palm gif

Thanks for catching this!

**Pull request merged**